packages/components/src/components/segmented-control/segmented-control-item/segmented-control-item-styles.ts

```diff
@@ -23,36 +28,47 @@ export const getColors = (
   isSelected: boolean,
   bgColor: SegmentedControlBackgroundColor,
   theme: Theme
-): { backgroundColor: string; buttonColor: string; labelColor: string } => {
+): {
+  backgroundColor: string;
+  buttonColor: string;
+  labelColor: string;
+  borderColor: string;
+  hoverBorderColor: string;
+} => {
   const themedColors = getThemedColors(theme);
-  const { primaryColor, contrastMediumColor } = isSelected ? getInvertedThemedColors(theme) : themedColors;
 
-  const backgroundColor =
-    themedColors[
-      isSelected ? 'contrastHighColor' : bgColor === 'background-surface' ? 'backgroundColor' : 'backgroundSurfaceColor'
-    ];
+  const { primaryColor, contrastMediumColor, disabledColor, contrastLowColor } = themedColors;
+  const backgroundColor = themedColors[bgColor === 'background-surface' ? 'backgroundColor' : 'backgroundSurfaceColor'];
+  const borderColor = isSelected ? primaryColor : contrastLowColor;
 
   return isDisabled
     ? {
         backgroundColor,
-        buttonColor: themedColors.disabledColor,
-        labelColor: themedColors.disabledColor,
+        buttonColor: disabledColor,
+        labelColor: disabledColor,
+        borderColor,
+        hoverBorderColor: primaryColor,
       }
     : {
         backgroundColor,
         buttonColor: primaryColor,
         labelColor: contrastMediumColor,
+        borderColor,
+        hoverBorderColor: primaryColor,
       };
 };
 
 export const getComponentCss = (
   isDisabled: boolean,
   isSelected: boolean,
   bgColor: SegmentedControlBackgroundColor,
-  theme: Theme
+  theme: Theme,
+  icon?: LinkButtonIconName,
```

Comment:
How can those be optional? Theme should be last.

Reply:
fixed it